### PR TITLE
Add permission for each flight mode in Flight Multiability

### PR DESCRIPTION
**Pull request by LucioMacarine**
## Additions
* Added permission checks for Soar, Glide and Levitate in the Flight MultiAbility

### CODE REVIEW

**Comment by LucioMacarine** [comment]
I'm new here so please point out anything that i might have done wrong. :heart:

**Review by Varhagna — APPROVED** [review]

File: src/com/projectkorra/projectkorra/airbending/flight/FlightMultiAbility.java [hunk 1]
```diff
@@ -47,6 +48,9 @@ private static enum FlightMode {
 		SOAR, GLIDE, LEVITATE, ENDING;
 	}
 
+	private Map<FlightMode, Boolean> allowedFlightModes = new HashMap<>();
```
Comment (StrangeOne101) [inline src/com/projectkorra/projectkorra/airbending/flight/FlightMultiAbility.java:51]: If you have a boolean for a value in a map, you may as well just use a Set. And use `Set#contains`

File: src/com/projectkorra/projectkorra/airbending/flight/FlightMultiAbility.java [hunk 2]
```diff
@@ -97,6 +101,15 @@ public FlightMultiAbility(final Player player) {
 			return;
 		}
 
+		this.allowedFlightModes.put(FlightMode.SOAR, player.hasPermission("bending.ability.flight.soar"));
+		this.allowedFlightModes.put(FlightMode.GLIDE, player.hasPermission("bending.ability.flight.glide"));
+		this.allowedFlightModes.put(FlightMode.LEVITATE, player.hasPermission("bending.ability.flight.levitate"));
+		this.firstAvaliableFlightMode = this.allowedFlightModes
+			.keySet()
+			.stream()
+			.filter(mode -> this.allowedFlightModes.get(mode)).findFirst();
```
Comment (StrangeOne101) [inline src/com/projectkorra/projectkorra/airbending/flight/FlightMultiAbility.java:110]: While I like what you are trying to achieve, I think this sort of thing should be implemented for multiabilities as a whole. Right now, WaterArms supports permissions for each subability, but they aren't removed from the list. So I think this functionality would be better suited to be fused into Multiabilities as a whole. And then WaterArms and flight are just updated to use the new system.

**Review by StrangeOne101 — CHANGES_REQUESTED** [review]
I like what you are trying to do. But I think the changes you are making should be made more in MultiAbility. See my comments bellow about it.

In addition, you are also using the `wip` branch of PK. This branch is pretty old and isn't updated now, so try do it from `master` to `master`

**Comment by StrangeOne101** [comment]
If the changes mentioned are too much, just copy how WaterArms does it for now while we figure out a better way to do it systematically

**Comment by LucioMacarine** [comment]
I'll work on it soon

**Comment by LucioMacarine** [comment]
As it turns out, the changes mentioned above were too much. ☹️
I deeply apologize but don't feel very confident changing that much. I tried to think of some ideas but they all had compromises and were just bad in general.

As for making it more like WaterArms, WaterArms only has to check the permission when a sub ability gets activated and not on startup that's why i added the first avaliable flight mode thing. So reverting to this PR's first commit would do exactly that i guess

And one more dumb question:
For me to be able to merge from master to master would i have to create another PR?

I truly apologize for wasting your times with this seemingly simple patch.

**Comment by StrangeOne101** [comment]
Don't apologize, these things happen.

I think you would have to create another PR to go from master to master.

**Pull request closed without merging**